lib/call-cabal-project-to-nix.nix

```diff
@@ -230,7 +230,11 @@ let
             else if repoData.sha256 != null
             then fetchgit { inherit (repoData) url sha256; rev = repoData.rev or repoData.ref; }
             else
-              let drv = builtins.fetchGit { inherit (repoData) url ; rev = repoData.rev or repoData.ref; ref = repoData.ref or null; };
+              let drv = builtins.fetchGit
+                { inherit (repoData) url ; ref = repoData.ref or null; }
+                # fetchGit does not accept "null" as rev, so when it's null
+                # we have to omit the argument completely.
+                // (if repoData.rev or false then { rev = repoData.rev; } else {});
```

Comment:
I think `if repoData.rev or false` will break when rev is present (because it is not a bool), maybe you wanted `if repData ? rev`.  However how about:
```
  // pkgs.lib.optionalAttrs (repoData ? rev) { inherit (repoData) rev; }
```

Reply:
Wow, thanks for catching that - I'm not too well familiar with Nix language. Will fix today.

I've also not known about `optionalAttrs`, will do that as well

Reply:
Fixed!